sw/host/opentitanlib/src/util/nix.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use nix::libc::{c_int, c_ulong};
+//use nix::poll;
+use nix::errno::errno;
+use nix::sys::signal::Signal;
+
+#[link(name = "c")]
+extern "C" {
+    fn prctl(option: c_int, arg2: c_ulong, arg3: c_ulong, arg4: c_ulong, arg5: c_ulong) -> c_int;
+}
+
+pub fn request_parent_death_signal(signal: Signal) -> Result<()> {
+    const PR_SET_PDEATHSIG: c_int = 1;
+    let res = unsafe { prctl(PR_SET_PDEATHSIG, signal as c_ulong, 0, 0, 0) };
+    if res != 0 {
+        Err(anyhow::anyhow!(
+            "prctl(PR_SET_PDEATHSIG): errno={}",
+            errno()
+        ))
```

Comment:
Can we replace this with a real error struct?  I think `nix::errno::Errno` implements `Error`, so we could probably do something like this:

```
use anyhow::Context;
use nix::errno::Errno;
...
if res != 0 {
  Err(Errno::last()).context("prctl(PR_SET_PDEATHSIG")
}
```

Reply:
These are excellent suggestions, done.